src/Adapters/Auth/huawei.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+const https = require('https');
+var Parse = require('parse/node').Parse;
+
+const TOKEN_ISSUER = "https://accounts.huawei.com";
+const HUAWEI_URL =   "https://oauth-login.cloud.huawei.com/oauth2/v3/tokeninfo";
+
+
+function getDataFromHuawei(id_token){
+  return new Promise((resolve, reject) => {
+    https.get(`${HUAWEI_URL}?id_token=${id_token}`, res => {
+      let data = '';
+      res.on('data', chunk => {
+        data += chunk.toString('utf8');
+      });
+      res.on('end', () => {
+        resolve(JSON.parse(data));
+      });
+    }).on('error', reject);
+  });
+}
+
+
+async function verifyIdToken({ id_token: token, id }) {
+  if (!token) {
+    throw new Parse.Error(Parse.Error.OBJECT_NOT_FOUND, `id token is invalid for this user.`);
+  }
+  const huaweiData = await getDataFromHuawei(token);
+  // Error from huawei
+  if(!huaweiData || huaweiData.error){
+    throw new Parse.Error(Parse.Error.OTHER_CAUSE,
+      huaweiData.error_description ?
+        `${huaweiData.error_description} (${huaweiData.error}::${huaweiData.sub_error})` :
+        `undefined Error from huawei (${huaweiData.error}::${huaweiData.sub_error || "No SubError"}})`
```

Comment:
Could you correct the capitalization to:
> Undefined error from Huawei API ...

and what does "No SubError" mean?

Reply:
If there is error, Huawei response JSON like this ( [Doc](https://developer.huawei.com/consumer/en/doc/development/HMSCore-Guides/open-platform-error-0000001053869182) )
{
    error: 1500,
    sub_error: 15006,
    error_description:" ... "
}
In error case 1203, there is no sub-error but I didn't experience that case
So I don't know sub_error field is undefined, empty string or zero.

Reply:
I remove ambiguous message "No SubError". and simplify code